include/alpaka/intrinsic/IntrinsicUniformCudaHipBuiltIn.hpp

```diff
@@ -56,27 +56,19 @@ namespace alpaka
                 //-----------------------------------------------------------------------------
                 __device__ static auto popcount(
                     intrinsic::IntrinsicUniformCudaHipBuiltIn const & /*intrinsic*/,
-                    unsigned int value)
-                -> int
+                    std::uint32_t value)
+                -> std::int32_t
                 {
-#if BOOST_COMP_CLANG
-                    return __popc(static_cast<int>(value));
-#else
-                    return __popc(value);
-#endif
+                    return __popc(static_cast<unsigned int>(value));
```

Comment:
I think here the `#if` guards should stay. The issue is that clang defines a different interface of `__popc` taking a signed int, while all other compilers have unsigned. Same for the other version of popcount.

Reply:
HIP and CUDA define the function with unsigned int.

CUDA is unsigned int: https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__INT.html#group__CUDA__MATH__INTRINSIC__INT_1g43c9c7d2b9ebf202ff1ef5769989be46
HIP is unsigned int: https://rocmdocs.amd.com/en/latest/Programming_Guides/HIP-GUIDE.html

Reply:
ohh no you are right clang cuda define it as `int` http://clang.llvm.org/doxygen/____clang__cuda__device__functions_8h.html#a2d0eb6e60af727a007e0086d27718517
Than I need to update the guard because hip-clang should use `unsigned`